src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/NativeMemoryTests.cs

```diff
@@ -558,5 +558,75 @@ public void ClearWithNullPointerAndZeroByteCountTest()
 
             // This test method just needs to check that no exceptions are thrown
         }
+
+        [Fact]
+        public void CopyNullBlockShouldNoOpTest()
+        {
+            // This should not throw
+            NativeMemory.Copy(null, null, 0);
+        }
+
+        [Fact]
+        public void CopyEmptyBlockShouldNoOpTest()
+        {
+            int* source = stackalloc int[1] { 42 };
+            int* destination = stackalloc int[1] { 0 };
+
+            NativeMemory.Copy(source, destination, 0);
+
+            Assert.Equal(0, destination[0]);
+        }
+
+        [Theory]
+        [InlineData(1, 1, 1)]
+        [InlineData(7, 9, 5)]
+        [InlineData(1, 16, 1)]
+        [InlineData(16, 16, 16)]
+        [InlineData(29, 37, 19)]
+        [InlineData(1024, 16, 16)]
+        public void CopyTest(int sourceSize, int destinationSize, int byteCount)
+        {
+            void* source = NativeMemory.AllocZeroed((nuint)sourceSize);
+            void* destination = NativeMemory.AllocZeroed((nuint)destinationSize);
+
+            new Span<byte>(source, sourceSize).Fill(0b10101010);
+
+            NativeMemory.Copy(source, destination, (nuint)byteCount);
+
+            Equals(byteCount - 1, new Span<byte>(destination, destinationSize).LastIndexOf<byte>(0b10101010));
+
+            NativeMemory.Free(source);
+            NativeMemory.Free(destination);
+        }
+
+        [Fact]
+        public void CopyToOverlappedMemoryTest()
+        {
+            byte* source = (byte*)NativeMemory.AllocZeroed(311);
+            Random.Shared.NextBytes(new Span<byte>(source, 50));
+
+            NativeMemory.Copy(source, source + 100, 50);
```

Comment:
this is copying only part of the overlapping buffer. what about a test for copying into the part of the buffer being read from? eg., remove the +100 or make it +10.

Reply:
Also should `NativeMemory.Copy(source, source, something)` no-op? or segfault eg if source+something is illegal?

Reply:
Done.